dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java

```diff
@@ -144,13 +209,75 @@ public void shouldSaveRuleAction() throws DotSecurityException, DotDataException
+        this.addRulesPermission(role, host, true);
+
+        final RuleAction ruleAction = new RuleActionDataGen().rule(rule).next();
+
+        rulesAPI.saveRuleAction(ruleAction, user, false);
+
+        final List<Rule> allRules = rulesAPI.getAllRules(user, false);
+        checkActions(ruleAction, allRules);
+    }
+
+    /**
+     * Method to Test: {@link RulesAPIImpl#saveRuleAction(RuleAction, User, boolean)}
+     * When: User with permission try to create a Rule's Action in a Rule's Page
+     * Should: Save it
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldSaveRuleActionInPage() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final Host host = new SiteDataGen().nextPersisted();
+
+         final Template template = new TemplateDataGen().nextPersisted();
+         final HTMLPageAsset htmlPageAsset = new HTMLPageDataGen(host, template)
+                .nextPersisted();
+
+        final Rule rule = new RuleDataGen().page(htmlPageAsset).nextPersisted();
 
+        this.addRulesPermission(role, host, true);
+        this.addPermission(role, htmlPageAsset, PermissionLevel.PUBLISH.getType());
         final RuleAction ruleAction = new RuleActionDataGen().rule(rule).next();
 
         rulesAPI.saveRuleAction(ruleAction, user, false);
 
         final List<Rule> allRules = rulesAPI.getAllRules(user, false);
+        checkActions(ruleAction, allRules);
+    }
+
+    /**
+     * Method to Test: {@link RulesAPIImpl#saveRuleAction(RuleAction, User, boolean)}
+     * When: User with permission try to create a Rule's Action but without PUBLISH permission over the page
+     * Should: throw a {@link DotSecurityException}
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test (expected = DotSecurityException.class)
+    public void shouldSaveRuleActionInPageShouldNotWork() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final Host host = new SiteDataGen().nextPersisted();
+
+        final Template template = new TemplateDataGen().nextPersisted();
+        final Folder folder = new FolderDataGen().site(host).nextPersisted();
+        final HTMLPageAsset htmlPageAsset = new HTMLPageDataGen(host, template)
+                .folder(folder)
+                .nextPersisted();
+
+        final Rule rule = new RuleDataGen().page(htmlPageAsset).nextPersisted();
+
+        this.addRulesPermission(role, host, true);
+        final RuleAction ruleAction = new RuleActionDataGen().rule(rule).next();
+
+        rulesAPI.saveRuleAction(ruleAction, user, false);
+    }
+
+    private void checkActions(RuleAction ruleAction, List<Rule> allRules) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'ruleAction' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)